python/dgl/heterograph.py

```diff
@@ -264,7 +264,7 @@ def __repr__(self):
                    '      metagraph={meta})')
             nnode_dict = {self.ntypes[i] : self._graph.number_of_nodes(i)
                           for i in range(len(self.ntypes))}
-            nedge_dict = {self.etypes[i] : self._graph.number_of_edges(i)
+            nedge_dict = {self.canonical_etypes[i] : self._graph.number_of_edges(i)
```

Comment:
Could we revert this? Changing this means we need to modify all the docstrings and tutorials.

Reply:
I can update the documentation accordingly. I find it quite confusing and misleading when there are repeated edge types as they will occur only once.

Reply:
Another concern is compactness. Maybe just print out the number of types if it's too long?

Reply:
I imagine users will not print it if it is too long and we do not need to care about that.

Reply:
I've updated the documentation to use `canonical_etypes`.